plugins/MonitorStage/MonitorMain.qml

```diff
@@ -2,20 +2,43 @@
 // Cura is released under the terms of the LGPLv3 or higher.
 
 import QtQuick 2.10
-import QtQuick.Controls 1.4
-
+import QtQuick.Controls 2.0
 import UM 1.3 as UM
 import Cura 1.0 as Cura
 
-
 // We show a nice overlay on the 3D viewer when the current output device has no monitor view
 Rectangle
 {
     id: viewportOverlay
 
+    property bool isConnected: Cura.MachineManager.activeMachineHasActiveNetworkConnection || Cura.MachineManager.activeMachineHasActiveCloudConnection
+    property bool isNetworkConfigurable: ["Ultimaker 3", "Ultimaker 3 Extended", "Ultimaker S5"].indexOf(Cura.MachineManager.activeMachineDefinitionName) > -1
+    property bool isNetworkConfigured:
+    {
+        // Readability:
+        var connectedTypes = [2, 3];
+        var types = Cura.MachineManager.activeMachineConfiguredConnectionTypes
+
+        // Check if configured connection types includes either 2 or 3 (LAN or cloud)
+        for (var i = 0; i < types.length; i++)
+        {
+            if (connectedTypes.indexOf(types[i]) >= 0)
+            {
+                return true
+            }
+        }
+        return false
+    }
```

Comment:
Isn't this code just a union of two sets with a test for non-zero?

Reply:
Nope. 1 = USB printing.